bookkeeper-server/src/main/java/org/apache/bookkeeper/http/ListBookieInfoService.java

```diff
@@ -0,0 +1,121 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.http;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.util.Map;
+import org.apache.bookkeeper.client.BookKeeper;
+import org.apache.bookkeeper.client.BookieInfoReader;
+import org.apache.bookkeeper.conf.ClientConfiguration;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.http.service.HttpService;
+import org.apache.bookkeeper.http.service.HttpServiceRequest;
+import org.apache.bookkeeper.http.service.HttpServiceResponse;
+import org.apache.bookkeeper.net.BookieSocketAddress;
+import org.apache.bookkeeper.util.JsonUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HttpService that handle Bookkeeper Configuration related http request.
+ */
+public class ListBookieInfoService implements HttpService {
+
+    static final Logger LOG = LoggerFactory.getLogger(ListBookieInfoService.class);
+
+    protected ServerConfiguration conf;
+
+    public ListBookieInfoService(ServerConfiguration conf) {
+        Preconditions.checkNotNull(conf);
+        this.conf = conf;
+    }
+
+    String getReadable(long val) {
+        String unit[] = {"", "KB", "MB", "GB", "TB" };
+        int cnt = 0;
+        double d = val;
+        while (d >= 1000 && cnt < unit.length - 1) {
+            d = d / 1000;
+            cnt++;
+        }
+        DecimalFormat df = new DecimalFormat("#.###");
+        df.setRoundingMode(RoundingMode.DOWN);
+        return cnt > 0 ? "(" + df.format(d) + unit[cnt] + ")" : unit[cnt];
+    }
+
+    @Override
+    public HttpServiceResponse handle(HttpServiceRequest request) throws Exception {
+        HttpServiceResponse response = new HttpServiceResponse();
+
+        if (HttpServer.Method.GET == request.getMethod()) {
+            ClientConfiguration clientConf = new ClientConfiguration(conf)
+              .setZkServers(conf.getZkServers());
+            clientConf.setDiskWeightBasedPlacementEnabled(true);
+            BookKeeper bk = new BookKeeper(clientConf);
+
+            Map<BookieSocketAddress, BookieInfoReader.BookieInfo> map = bk.getBookieInfo();
+            if (map.size() == 0) {
+                bk.close();
+                response.setCode(HttpServer.StatusCode.NOT_FOUND);
+                response.setBody("Not found any Bookie info.");
+                return response;
+            }
+
+            /**
+             * output:
+             *  {
+             *    "bookieAddress" : "free: xxx \t total: xxx \n",
```

Comment:
since you are returning a json object, why not make the returned value structured?
```
bookieAddress : { "free" : 12345, "total": 34567 }
```

Reply:
Thanks. will change it.